sdk/sdktypes/var.go

```diff
@@ -23,11 +23,24 @@ func (p Var) ScopeID() VarScopeID { return kittehs.Must1(ParseVarScopeID(p.read(
 func (p Var) Name() Symbol        { return kittehs.Must1(ParseSymbol(p.read().Name)) }
 func (p Var) Value() string       { return p.read().Value }
 func (p Var) IsSecret() bool      { return p.read().IsSecret }
+func (p Var) IsRequired() bool    { return p.read().IsRequired }
 
 func (p Var) WithScopeID(id VarScopeID) Var {
 	return Var{p.forceUpdate(func(pb *VarPB) { pb.ScopeId = id.String() })}
 }
 
-func NewVar(n Symbol, v string, isSecret bool) Var {
-	return kittehs.Must1(StrictVarFromProto(&VarPB{Name: n.String(), Value: v, IsSecret: isSecret}))
+func (p Var) SetSecret(s bool) Var {
```

Comment:
This works, but that fact that it works on a value receiver is confusing.

Reply:
Also, I don't like this builder pattern :)

Reply:
i know you hate builders :)